File: wallet/src/main/java/com/worldventures/wallet/ui/wizard/input/scanner/WizardScanBarcodePresenter.java
package com.worldventures.wallet.ui.wizard.input.scanner;

import com.worldventures.wallet.ui.common.base.WalletPresenter;

public interface WizardScanBarcodePresenter extends WalletPresenter<WizardScanBarcodeScreen> {

   void goBack();

   void requestCamera();

   void barcodeScanned(String barcode);

   void startManualInput();

   void retry(String barcode);

   void retryAssignedToCurrentDevice();

   void retryScan();
}

File: wallet/src/main/java/com/worldventures/wallet/ui/wizard/input/scanner/WizardScanBarcodePresenter.kt
package com.worldventures.wallet.ui.wizard.input.scanner

import com.worldventures.wallet.ui.common.base.WalletPresenter
import com.worldventures.wallet.ui.wizard.input.helper.BaseBarcodeInputPresenter

interface WizardScanBarcodePresenter : WalletPresenter<WizardScanBarcodeScreen>, BaseBarcodeInputPresenter {

   fun goBack()

   fun requestCamera()

   fun startManualInput()

   fun retryScan()
}
